cmd/telegraf/telegraf.go

```diff
@@ -231,30 +232,61 @@ func (t *Telegraf) watchLocalConfig(ctx context.Context, signals chan os.Signal,
 		return
 	}
 	log.Printf("I! Config watcher started for %s\n", fConfig)
-	select {
-	case <-ctx.Done():
-		mytomb.Done()
-		return
-	case <-changes.Modified:
-		log.Printf("I! Config file/directory %q modified\n", fConfig)
-	case <-changes.Deleted:
-		// deleted can mean moved. wait a bit a check existence
-		<-time.After(time.Second)
-		if _, err := os.Stat(fConfig); err == nil {
-			log.Printf("I! Config file/directory %q overwritten\n", fConfig)
-		} else {
-			log.Printf("W! Config file/directory %q deleted\n", fConfig)
+
+	// Setup debounce timer
+	reloadTimer := time.NewTimer(t.watchDebounceInterval)
+	reloadTimer.Stop() // stop immediately to avoid triggering reload
+	reloadPending := false
+
+	// Helper to reset the timer and mark reload as pending
+	resetTimer := func(reason string) {
+		if !reloadPending {
+			reloadPending = true
+		}
+		reloadTimer.Reset(t.watchDebounceInterval)
+		log.Printf("%s", reason)
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			reloadTimer.Stop()
+			mytomb.Done()
+			return
+
+		case <-changes.Modified:
+			resetTimer(fmt.Sprintf("I! Config file/directory %q modified\n", fConfig))
+
+		case <-changes.Deleted:
+			// deleted can mean moved. wait a bit a check existence
+			<-time.After(time.Second)
+			var reason string
+			if _, err := os.Stat(fConfig); err == nil {
+				reason = fmt.Sprintf("I! Config file/directory %q overwritten\n", fConfig)
+			} else {
+				reason = fmt.Sprintf("W! Config file/directory %q deleted\n", fConfig)
+			}
+			resetTimer(reason)
```

Comment:
Unconditional blocking wait: This blocks the entire goroutine for 1 second, preventing handling of other events (like ctx.Done()) during this period
```suggestion
		    case <-changes.Deleted:
                              // Use select with timeout instead of blocking wait
                              timer := time.NewTimer(time.Second)
                              select {
                              case <-timer.C:
                                  // Proceed with file existence check
                              case <-ctx.Done():
                                  timer.Stop()
                                  return
                              }
                              
                              var reason string
                              if _, err := os.Stat(fConfig); err == nil {
                                  reason = fmt.Sprintf("I! Config file/directory %q overwritten\n", fConfig)
                              } else {
                                  reason = fmt.Sprintf("W! Config file/directory %q deleted\n", fConfig)
                              }
                              resetTimer(reason)
```